packages/itwin/grouping-mapping-widget/src/components/SharedComponents/utils.tsx

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { ProgressRadial, Text } from "@itwin/itwinui-react";
+import "./utils.scss";
+import React from "react";
+import type { IModelConnection } from "@itwin/core-frontend";
+
+export const handleInputChange = <T,>(
+  e: React.ChangeEvent<HTMLInputElement>,
+  values: T,
+  setValues: (newValues: T) => void,
+) => {
+  const { name, value } = e.target;
+
+  setValues({
+    ...values,
+    [name]: value,
+  });
+};
+
+export const LoadingOverlay = () => (
```

Comment:
These should be separate components in their separate files:
`LoadingOverlay`
`EmptyMessage`
`LoadingSpinner`